### PR TITLE
rgw: return error if lock log shard fails

### PR DESCRIPTION
**Pull request by zhangsw**
Fixes: https://tracker.ceph.com/issues/43453

Signed-off-by: zhangshaowen <zhangshaowen@cmss.chinamobile.com>

### CODE REVIEW

File: src/rgw/rgw_reshard.cc [hunk 1]
```diff
@@ -993,6 +993,9 @@ int RGWReshard::process_single_logshard(int logshard_num)
       logshard_oid << dendl;
     return ret;
   }
+  if (ret < 0) {
+    return ret;
+  }
```
Comment (cbodley) [inline src/rgw/rgw_reshard.cc]: I think we should just replace the `ret == -EBUSY` above with `ret < 0`

**Comment by cbodley** [comment]
@zhangsw can you please open a tracker issue for backports?

File: src/rgw/rgw_reshard.cc (hunk 1 above)
Comment (ivancich) [reply src/rgw/rgw_reshard.cc]: And maybe also indicate the error code in the logged message, so we can better understand future failures.

**Comment by cbodley** [comment]
jenkins test make check

**Comment by cbodley** [comment]
http://pulpito.ceph.com/cbodley-2020-01-21_20:23:35-rgw-wip-cbodley-testing-distro-basic-smithi/

**Pull request merged**